pkgs/leak_tracker/lib/src/leak_tracking/_leak_filter.dart

```diff
@@ -28,24 +34,35 @@ class _PhaseLeakFilter {
   final PhaseSettings phase;
 
   bool shouldReport(LeakType leakType, ObjectRecord record) {
+    final bool result;
     switch (leakType) {
       case LeakType.notDisposed:
-        return _shouldReport(
+        result = _shouldReportByTypeAndClass(
           leakType,
           record,
           phase.ignoredLeaks.notDisposed,
         );
       case LeakType.notGCed:
       case LeakType.gcedLate:
-        return _shouldReport(
+        result = _shouldReportByTypeAndClass(
           leakType,
           record,
           phase.ignoredLeaks.notGCed,
         );
     }
+
+    // Check for test helpers should happen only in case of leak because
+    // it is performance heavy.
+    final shouldCheckCreator =
+        phase.ignoredLeaks.createdByTestHelpers && result;
+
+    if (!shouldCheckCreator) return result;
+
+    return !(record.creationChecker?.createdByTestHelpers ?? false);
   }
 
-  bool _shouldReport(
+  /// Returns true if the leak should not be ignored because of type or class.
```

Comment:
nit: remove the double negative "not be ignored". 

"Returns whether the leak should be reported based on its type and class."